utils/connect.go

```diff
@@ -10,34 +10,43 @@ import (
 	"github.com/BoltzExchange/boltz-client/logger"
 )
 
-func ConnectBoltz(lightning lightning.LightningNode, boltz *boltz.Boltz) (string, error) {
-	nodes, err := boltz.GetNodes()
+func connect(node lightning.LightningNode, peer *boltz.NodeInfo) error {
+	if len(peer.Uris) == 0 {
+		return fmt.Errorf("no uris found for peer: %s", peer.PublicKey)
+	}
+	uri := peer.Uris[0]
+	err := node.ConnectPeer(uri)
+	if err == nil {
+		logger.Infof("Connected to Boltz node: %s", uri)
+	} else if strings.Contains(err.Error(), "already connected to peer") {
+		logger.Infof("Already connected to Boltz node: %s", uri)
+		err = nil
+	} else {
+		err = fmt.Errorf("could not connect to boltz node %s: %w", uri, err)
+	}
 
+	return err
+}
+
+func ConnectBoltz(lightning lightning.LightningNode, boltz *boltz.Boltz) error {
+	nodes, err := boltz.GetNodes()
 	if err != nil {
-		return "", err
+		return err
 	}
 
 	symbol := "BTC"
-	node, hasNode := nodes.Nodes[symbol]
+	btcNodes, hasNode := nodes[symbol]
 
 	if !hasNode {
-		return "", errors.New("could not find Boltz node for symbol: " + symbol)
+		return errors.New("could not find Boltz node for symbol: " + symbol)
 	}
 
-	if len(node.URIs) == 0 {
-		return node.NodeKey, errors.New("could not find URIs for Boltz LND node for symbol: " + symbol)
+	if err := connect(lightning, btcNodes.LND); err != nil {
+		return err
 	}
-	uri := node.URIs[0]
-	err = lightning.ConnectPeer(uri)
 
-	if err == nil {
-		logger.Info("Connected to Boltz node: " + uri)
-	} else if strings.Contains(err.Error(), "already connected to peer") {
-		logger.Info("Already connected to Boltz node: " + uri)
-		err = nil
-	} else {
-		err = fmt.Errorf("could not connect to boltz node %s: %w", uri, err)
+	if err := connect(lightning, btcNodes.CLN); err != nil {
```

Comment:
The `LND` and the `CLN` fields are both optional. Make sure there is no error thrown when the backend is running without one of those two